Source/Parsing/Frame/Content/Adapter/ID3TableOfContentsStringAdapter.swift

```diff
@@ -42,16 +48,20 @@ class ID3TableOfContentsStringAdapter: TableOfContentsStringAdapter {
         }
     }
 
-    private func getEntryCountFrom(contentComponents: [String]) -> Int {
-        return Int(contentComponents[3]) ?? 0
+    private func getEntryCountFrom(contentComponents: [String]) -> UInt8 {
+        #warning("Is falling back to zero really the right thing to do?")
+        #warning("Can 3 be out of bounds? Is trapping in this case okay?")
```

Comment:
Again, sorry for the delay in responding. GitHub didn't notify me that you'd added the branch so I didn't know this was here until this afternoon (I was wondering why nothing had shown up after your comment to me yesterday) and haven't had a chance to look at all your comments until tonight.

I'm not sure if falling back to zero is the right thing to do. Ideally, if you have a TOC, it's doing to have something in it, right? There should be at least one chapter or sub-TOC?

As for 3 being out of bounds...I don't think so. Of these five parameters, only the sub-frame (5) is optional. If you're going to have a TOC frame, the first four have to be there. But if there's a more bullet-proof way to handle this than referencing the index of the parameter in the array of parameters, I'm all for it.

Reply:
That was a note for myself to come back and figure out where the array originated from. It doesn’t have anything to do with the file specification. If the array was assembled in code in a way that guarantees a certain number of elements, than these indices can be safely accessed like this. But if the array’s length is derived from what it finds in the file, then a corrupt file could lead to a shorter‐than‐expected array, and when you ask for an index that is out of bounds, the application will simply crash. There are a lot of better options, such as throwing, or ignoring the entire frame. But I will come back to look at these sorts of nuances after I have gotten the normal program flow to work with a valid file.